frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BaseRenameAction.java

```diff
@@ -91,15 +92,20 @@ private static void rename(Project project,
     val options = gitNewBranchDialog.showAndGetOptions();
 
     if (options != null) {
-      GitBrancher brancher = GitBrancher.getInstance(project);
-      Runnable renameRunnable = () -> brancher.renameBranch(currentBranchName, options.getName(),
-          Collections.singletonList(gitRepository)); // runs in background
-      new RenameBackgroundable(gitRepository,
-          branchLayout,
-          branchLayoutWriter,
-          renameRunnable,
-          currentBranchName,
-          options.getName()).queue();
+      Path macheteFilePath = gitRepository.getMacheteFilePath();
+      val newBranchLayout = branchLayout.rename(currentBranchName, options.getName());
+
+      runWriteActionOnUIThread(() -> {
+        MacheteFileWriter.writeBranchLayout(
+            macheteFilePath,
+            branchLayoutWriter,
+            newBranchLayout,
+            /* backupOldLayout */ true,
+            /* requestor */ this);
```

Comment:
Hmmm what's `this` in this context? That's probably the lambda object... is it relevant?

Reply:
its `RenameSelectedAction` similarly to `DiscoverAction` - I think we are good